src/shared/filesystemUtilities.ts

```diff
@@ -198,18 +196,23 @@ export function getFileDistance(fileA: string, fileB: string): number {
  * @param suffix  Filename suffix, typically an extension (".txt"), may be empty
  * @param max  Stop searching if all permutations up to this number exist
  */
-export function getNonexistentFilename(dir: string, name: string, suffix: string, max: number = 99): string {
+export async function getNonexistentFilename(
+    dir: string,
+    name: string,
+    suffix: string,
+    max: number = 99
+): Promise<string> {
     if (!name) {
         throw new Error(`name is empty`)
     }
-    if (!fs.existsSync(dir)) {
+    if (!(await fsCommon.directoryExists(dir))) {
```

Comment:
"dir" is very commonly used in our (and nodejs's) existing filesystem code. No need for the verbose form.
```suggestion
    if (!(await fsCommon.dirExists(dir))) {
```